source/gui.cpp

```diff
@@ -1786,6 +1786,139 @@ void reshade::runtime::draw_overlay_variable_editor()
 
 			if (current_tree_is_open)
 			{
+				const float button_size = ImGui::GetFrameHeight();
+				const float button_spacing = _imgui_context->Style.ItemInnerSpacing.x;
+
+				ImGui::PushStyleVar(ImGuiStyleVar_FramePadding, ImVec2(_imgui_context->Style.FramePadding.x, 0));
+				if (imgui_popup_button("Edit effect preprocessor definitions", -1, ImGuiWindowFlags_NoMove))
+				{
+					ImGui::PopStyleVar();
+
+					std::vector<std::string> user_definitions;
```

Comment:
This should be `std::set` to avoid duplicates (`set` instead of `unordered_set` to guarantee ordering in the UI).